account/manager.go

```diff
@@ -1630,38 +1652,69 @@ func (m *manager) createSpendTx(ctx context.Context, account *Account,
 		return nil, err
 	}
 
-	// Gather the remaining components required to sign the transaction
-	// fully. This includes signing the account input and any additional
-	// ones.
-	sigHashType := txscript.SigHashAll
-	for i, txIn := range tx.TxIn {
-		if i == accountInputIdx {
-			continue
-		}
+	// Our account input signature isn't always all that's required to spend
+	// it, so we'll take care of forming a proper signature later.
+	err = m.decorateAccountInput(account, packet, accountInputIdx)
+	if err != nil {
+		return nil, err
+	}
 
-		inputScript, err := m.signInput(
-			ctx, tx, inputMap[txIn.PreviousOutPoint], i,
-			sigHashType,
-		)
-		if err != nil {
-			return nil, err
-		}
+	// Let the wallet sign each input. This will add a partial signature for
+	// the account input which we'll later turn into the correct witness
+	// depending on the spend path.
+	signedPacket, err := m.cfg.Wallet.SignPsbt(ctx, packet)
+	if err != nil {
+		return nil, err
+	}
+
+	pIn := &signedPacket.Inputs[accountInputIdx]
+	witnessScript := pIn.WitnessScript
 
-		txIn.SignatureScript = inputScript.SigScript
-		txIn.Witness = inputScript.Witness
+	if len(pIn.PartialSigs) != 1 {
+		return nil, fmt.Errorf("unexpected number of signatures in "+
+			"signed PSBT, got %d wanted 1", len(pIn.PartialSigs))
 	}
 
-	// Our account input signature isn't always all that's required to spend
-	// it, so we'll take care of forming a proper signature later.
-	witnessScript, ourSig, err := m.signAccountInput(
-		ctx, tx, account, accountInputIdx, sigHashType,
-	)
+	ourSig := pIn.PartialSigs[0].Signature
+
+	// We temporarily set the final witness to the partial sig to allow
+	// extraction of the final TX.
+	pIn.FinalScriptWitness, err = serializeWitness([][]byte{ourSig})
```

Comment:
Yeah, the signing step only adds a raw signature to the PSBT but we need to create the full witness stack, which might also require us to call the auctioneer for its sig. So here we just add something to the final script witness so we can convert the PSBT into a normal TX (which does the work for us in terms of assembling the signatures for the other inputs in case of a deposit).